### PR TITLE
Updating README examples

### PR DESCRIPTION
**Pull request by evanjbowling**
First stab at #32

I'm also thinking:
- a javadoc link should be near the top
- the examples would be best as a set of files in the repo (or at least with `import` statements included in the README)
  - this would make it easier for users regardless of the IDE they use
- a dedicated Getting Started section (wiki?)

### CODE REVIEW

**Comment by evanjbowling** [comment]
NOTE: I didn't get through the Functors and ADTs sections yet.

File: README.md [hunk 1]
```diff
@@ -106,7 +106,7 @@ We could have also used `unfoldr`:
 ```Java
 Iterable<Integer> positiveSquaresBelow100 = unfoldr(x -> {
               int square = x * x;
-              return square < 100 ? Optional.of(tuple(square, x + 1)) : Optional.empty();
+              return square < 100 ? Maybe.maybe(tuple(square, x + 1)) : Maybe.nothing();
```
Comment (jnape) [inline README.md]: Let's use `Maybe.just` here for the positive case in the ternary switch; that's analogous to `Optional.of`

File: README.md [hunk 2]
```diff
@@ -144,7 +144,7 @@ And have fun with 3s:
 
 ```Java
 Iterable<Iterable<Integer>> multiplesOf3InGroupsOf3 =
-          take(3, inGroupsOf(3, unfoldr(x -> Optional.of(tuple(x * 3, x + 1)), 1)));
+          take(3, inGroupsOf(3, unfoldr(x -> Maybe.maybe(tuple(x * 3, x + 1)), 1)));
```
Comment (jnape) [inline README.md]: `Maybe.just` is what we want here, too

**Comment by jnape** [comment]
Yes, a javadoc link somewhere in there would definitely make sense.

I think your intuitions about Getting Started / examples is correct; at some point the tension of "explaining everything via a markdown file" vs "offering the most helpful information from scratch" is going to come to a head, at which point a micro-site is probably in order. In all likelihood, it's already time for that. That's a much bigger lift than just getting the README examples back to compiling, but is certainly worth doing at some point, and if you have interest in that, I'd be happy to talk about it.

**Comment by evanjbowling** [comment]
Made the changes you suggested - I think it's good to go now.

Since this is a small set of changes I chose to commit amend and rebase off palatable:master.

File: README.md [hunk 3]
```diff
@@ -125,9 +125,9 @@ Let's compose two functions:
 Fn1<Integer, Integer> add = x -> x + 1;
 Fn1<Integer, Integer> subtract = x -> x -1;
 
-Fn1<Integer, Integer> noOp = add.andThen(subtract);
+Fn1<Integer, Integer> noOp = add.fmap(subtract);
 // same as
-Fn1<Integer, Integer> alsoNoOp = subtract.compose(add);
+Fn1<Integer, Integer> alsoNoOp = subtract.fmap(add);
```
Comment (jnape) [inline README.md]: Whoops, I missed this in the first commit. This `fmap` should be `contraMap` to preserve the same order of operations (`add` then `subtract`) as the example immediately above it. I'll accept the PR and make the change on my end.

**Comment by jnape** [comment]
Merged in 02e350bd

**Pull request closed without merging**